File: inventory-management/src/main/java/com/inventory_management/service/CategoryService.java
package com.inventory_management.service;

import com.inventory_management.dto.request.CategoryRequestDTO;
import com.inventory_management.dto.response.CategoryResponseDTO;

import org.springframework.data.domain.Page;


public interface CategoryService {


    CategoryResponseDTO createCategory(CategoryRequestDTO request);


    Page<CategoryResponseDTO> getAllCategories(
            Boolean active,
            int page,
            int size,
            String sortBy,
            String sortDir
    );


    Page<CategoryResponseDTO> searchCategories(
            String keyword,
            Boolean active,
            int page,
            int size,
            String sortBy,
            String sortDir
    );


    CategoryResponseDTO getCategoryById(Integer categoryId);


    CategoryResponseDTO updateCategory(
            Integer categoryId,
            CategoryRequestDTO request
    );


    void deleteCategory(Integer categoryId);


    CategoryResponseDTO restoreCategory(Integer categoryId);

}

File: inventory-management-frontend/src/services/categoryService.js
import axiosInstance from "../api/axiosConfig";


const categoryService = {

    // Used by Product dropdown
    async getCategories(){

        const response = await axiosInstance.get(
            "/categories",
            {
                params:{
                    active:true,
                    page:0,
                    size:100,
                    sortBy:"categoryId",
                    sortDir:"asc"
                }
            }
        );


        return response.data.content;

    },

    // Active / Inactive listing

    async getAllCategories(params){

        const response = await axiosInstance.get(
            "/categories",
            {
                params
            }
        );


        return response.data;

    },

    // Search

    async searchCategories(params){

        const response = await axiosInstance.get(
            "/categories/search",
            {
                params
            }
        );


        return response.data;

    },

    // Create

    async createCategory(data){

        const response = await axiosInstance.post(
            "/categories",
            data
        );


        return response.data;

    },

    // Update

    async updateCategory(id,data){

        const response = await axiosInstance.put(
            `/categories/${id}`,
            data
        );


        return response.data;

    },

    // Soft delete

    async deleteCategory(id){

        await axiosInstance.delete(
            `/categories/${id}`
        );

    },


    // Restore

    async restoreCategory(id){

        await axiosInstance.patch(
            `/categories/${id}/restore`
        );

    }



};


export default categoryService;